### PR TITLE
Suggest picking a category on the pre-publish panel

### PR DESCRIPTION
**Pull request by youknowriad**
Before publishing a post, if it has no assigned category or has the "uncategorized" one, this PR shows a new suggestion on the panel to pick a category before publishing.

Screen Shot 2022-01-04 at 10 11 57 AM

### CODE REVIEW

File: packages/editor/src/components/post-publish-panel/maybe-category-panel.js [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+				categoriesTaxonomy.rest_base
+			);
+
+		return (
+			categoriesTaxonomy !== undefined &&
+			defaultCategory &&
+			some( categoriesTaxonomy.types, ( type ) => type === postType ) &&
+			( categories?.length === 0 ||
+				( categories?.length === 1 &&
+					defaultCategory.id === categories[ 0 ] ) )
```
Comment (Mamaduka) [inline packages/editor/src/components/post-publish-panel/maybe-category-panel.js]: This condition is a little hard to parse. Maybe we can split it into multiple returns since we're dealing with booleans?

**Review by Mamaduka — COMMENTED** [review]
Uses can change default post category from Settings > Writing. Do we want to handle this scenario as well?

**Comment by youknowriad** [comment]
> Uses can change default post category from Settings > Writing. Do we want to handle this scenario as well?

How do you think this would impact the behavior here?

**Comment by Mamaduka** [comment]
I think that depends on how we look at this feature.

A. If we want to suggest assigning another category than "default," we need to take user-defined settings into account.
B. If we only care about no category or "uncategorized," no need for additional changes.

**Comment by youknowriad** [comment]
It seems B is a better option to me, A default category (other than uncategorized) is probably always going to be a "good" category to pick from.

**Comment by Mamaduka** [comment]
We can also keep it as it is and improve when we need it.

**Comment by annezazu** [comment]
My understanding is that this PR will help folks pick a category other than Uncategorized but, if there's not another category, the only option once more is Uncategorized which defeats the purpose a bit IMO:
Screen Shot 2022-01-04 at 12 46 37 PM

**Comment by kellychoffman** [comment]
Nice! Would be cool if we also added a bit of educational text similar to tags. Maybe something like: 

> Categories provide a helpful way to group related posts together and to quickly tell readers what a post is about.

Screen Shot 2022-01-04 at 7 29 50 PM

**Comment by youknowriad** [comment]
> My understanding is that this PR will help folks pick a category other than Uncategorized but, if there's not another category, the only option once more is Uncategorized which defeats the purpose a bit IMO:

I thought about this but I decided to still keep the panel because they can add a category inline there.

> Nice! Would be cool if we also added a bit of educational text similar to tags. Maybe something like:

👍

**Comment by jameskoster** [comment]
This is neat, I often forget to assign categories :)

Do we definitely need the logic though... or would it be adequate to simply repeat the panels from the document inspector 1:1?

I ask because by comparison, I'm not sure that the "Suggestion: " text stands out enough to warrant the special treatment, and as mentioned above can feel a bit awkward in some circumstances. For example I don't use tags at all on my blog, and would find the constant "suggestion" a little aggravating, like my site doesn't know me :D 

Since the pre-publish panel is about double-checking settings, I don't know if we really need to frame these as "suggestions".

File: packages/editor/src/components/post-publish-panel/maybe-category-panel.js (hunk 1 above)
Comment (youknowriad) [reply packages/editor/src/components/post-publish-panel/maybe-category-panel.js]: I added a comment and extracted one part of the condition to clarify it.

**Comment by youknowriad** [comment]
I've added the educational text. If folks agree to remove "Suggestion:" from both tags and categories, I can do so 🤷‍♂️

File: packages/editor/src/components/post-publish-panel/maybe-category-panel.js (hunk 1 above)
Comment (Mamaduka) [reply packages/editor/src/components/post-publish-panel/maybe-category-panel.js]: Thanks.

**Review by Mamaduka — APPROVED** [review]
It looks good from a code perspective and as a feature.

I'm going to leave text copy decisions to others :)

**Comment by kellychoffman** [comment]
I think for the purpose and context of tis PR, let's keep 'suggestion'. It's straight forward in what it is and inline with tags. For folks like Jay, who don't utilize tags or cats, they can ignore the said suggestions - no harm, no foul. 

:shipit:
